controllers/disruption_controller.go

```diff
@@ -260,6 +260,14 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{Requeue: false}, nil
 		}
 
+		// check if we have reached trigger.createPods. If not, skip the rest of reconciliation.
+		requeueAfter := time.Until(TimeToCreatePods(instance.Spec.Triggers, instance.CreationTimestamp.Time))
+		if requeueAfter > 0 {
+			r.log.Debugw("requeuing disruption as we haven't yet reached trigger.createPods", "requeueAfter", requeueAfter.String())
+
+			return ctrl.Result{Requeue: false, RequeueAfter: requeueAfter}, nil
```

Comment:
```suggestion
			return ctrl.Result{RequeueAfter: requeueAfter}, nil
```[
RequeueAfter imply](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/reconcile/reconcile.go#LL32C31-L32C31) `requeue: true`